storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -607,6 +608,27 @@ protected String javaCmd(String cmd) {
         return ret;
     }
 
+    /**
+     * Extracting out to mock it for tests.
+     * @return true if on Linux.
+     */
+    protected boolean isOnLinux() {
+        return SystemUtils.IS_OS_LINUX;
+    }
+
+    private void prefixNumaPinningIfApplicable(String numaId, List<String> commandList) {
+        if (numaId != null) {
+            if (isOnLinux()) {
+                commandList.add("numactl");
+                commandList.add("--i=" + numaId);
+                return;
+            } else {
+                // TODO : Add support for pinning on Windows host
+                throw new RuntimeException("numactl pinning currently not supported on non-Linux hosts");
```

Comment:
Maybe `UnsupportedOperationException` is a better option. But I am fine with this.